### DISCUSSION TITLE
UseEslintrc is true，Error reporting：“code":"MODULE_NOT_FOUND”

### DISCUSSION THREAD

**Discussion by parlay96** [eslint/eslint#17377] [Q&A]
### Environment

Node version: 16.xxx
npm version: 6.14.16
Local ESLint version:  8.xx
Global ESLint version: 
Operating System: 

### What parser are you using?

Default (Espree)

### What did you do?

```
  private initESLintInstance(fix: boolean) {
    const eslint = new ESLintBase({
      cache: false, // 缓存
     // baseConfig: this.config, // 配置扩展
      cwd: this.directory, // 工作目录。这必须是一个绝对路径。
      fix, // 自动修复模式
      useEslintrc: true, // 默认为 true。如果 false，ESLint 不会加载配置文件（.eslintrc.* 文件）。只有构造函数选项的配置是有效的
    });

    return eslint;
  }
```

  ```
public async scan() {
    // 如果没有文件，直接返回空 检测结果
    if (this.targetFiles.length === 0) {
      return {
        data: [],
      };
    }

    try {
      const eslint = this.initESLintInstance(false);
      logger.log(1)
      // 返回检测结果
      const data = await eslint.lintFiles(this.targetFiles);
      logger.log(2)
      return {
        data,
      };
    } catch (e: any) {
      logger.log(e, 3)
      return { error: true, msg: 'eslint configuration exception, Could be a MODE_ NOT_ FOUND' }
    }
  }
```

` const results = await eslint.scan();  Error reporting`

### What did you expect to happen?

How to modify this issue？
options. resolvePluginsRelativeTo？What exactly does it mean？？
![image](https://github.com/eslint/eslint/assets/32978177/0e0e1d28-5fab-44b2-9712-96aaf92986c3)

### What actually happened?

![image](https://github.com/eslint/eslint/assets/32978177/a9144bfc-5c0a-4111-9f6c-85d2d9a19667)
![image](https://github.com/eslint/eslint/assets/32978177/059d4fcc-4b8c-4a17-a084-f907cf6e8a9c)

### Link to Minimal Reproducible Example

NO LINK

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by parlay96** [comment]
https://eslint.org/docs/latest/integrate/nodejs-api   I want to use the node API，no CLI， switch to a GUI, and implement my own detection. Currently, issues have been found with the configuration of eslintrc.js

**Comment by parlay96** [comment]
Is there anyone who knows about this problem？Thank you

**Comment by aladdin-add** [comment]
can you print the full error message (including the stack info)? and providing a repro can be helpful. :)

**Reply by parlay96** [reply]
@aladdin-add 
This is a vsocde plugin I developed to test the code quality of a project!
![image](https://github.com/eslint/eslint/assets/32978177/a80f4208-a988-4bab-a324-b45a9b913178)

This is how I used the vscode plugin in the project
![image](https://github.com/eslint/eslint/assets/32978177/ad65f774-41b5-4d0c-aa37-ddf9a3d1aeef)

I want to test the code for the project using eslint. 
`    const eslint = new ESLintBase({
      cache: false, // 缓存
     // baseConfig: this.config, // 配置扩展
      cwd: this.directory, // 工作目录。这必须是一个绝对路径。
      fix, // 自动修复模式
      resolvePluginsRelativeTo: this.directory,
      useEslintrc: true, // 默认为 true。如果 false，ESLint 不会加载配置文件（.eslintrc.* 文件）。只有构造函数选项的配置是有效的
    });
  this.directory:  It's the path of my current project, and I don't know where the problem caused me to report an error during testing
`
![image](https://github.com/eslint/eslint/assets/32978177/f8292d6c-34c6-4152-b82e-2a847333f91f)

![image](https://github.com/eslint/eslint/assets/32978177/54ef2005-53f5-4572-ac72-3a28c7988db6)

**Reply by aladdin-add** [reply]
I'm not an expert on vscode plugins, and it seems not an issue in eslint. you may need to look to see if vscode has a debugging mode to help find out what's going on.

**Reply by parlay96** [reply]
Thank you

**Comment by parlay96** [comment]
![image](https://github.com/eslint/eslint/assets/32978177/acc6f964-41fe-4da7-887b-342ba74e9c83)
